### PR TITLE
Process events

### PR DESCRIPTION
**Pull request by coloursofnoise**
## Summary
Add Sphinx events to post-process generated command documentation.

## Tasks

- [x] Added unit tests
- [x] Added documentation for new features (where applicable)
- [x] Added release notes (using [`reno`](https://pypi.org/project/reno/))
- [x] Ran test suite and style checks and built documentation (`tox`)

### CODE REVIEW

File: docs/usage.rst [hunk 1]
```diff
@@ -92,6 +92,28 @@ Programs
...
+
+   :param app: the Sphinx application object
+   :param ctx: the ``click.Context`` object used to generate the description
+   :param lines: the lines of the documentation, see below
+
+Events are emitted when sphinx-click has read and processed part of a
+command's documentation. *lines* is a list of strings -- the lines of the
+documentation that was processed -- that the event handler can
+modify **in place** to change what Sphinx puts into the output
+
```
Comment (stephenfin) [inline docs/usage.rst:115]: Could we get a small "hello, world"-like example of using this?

File: sphinx_click/ext.py [hunk 2]
```diff
@@ -25,6 +25,22 @@
 ANSI_ESC_SEQ_RE = re.compile(r'\x1B\[\d+(;\d+){0,2}m', flags=re.MULTILINE)
 
 
+def _process_lines(event_name):
+    def decorator(func):
```
Comment (stephenfin) [inline sphinx_click/ext.py:29]: Can we add [`functools.wraps`](https://docs.python.org/3/library/functools.html#functools.wraps) to preserve the docstrings etc.? Also, type annotations?

**Review by stephenfin — CHANGES_REQUESTED** [review]
Looks reasonable to me. Two small nits inline but otherwise I'm a-okay with this

**Comment by stephenfin** [comment]
Just the style issues to fix now. You can do this by installing and running pre-commit.

```
pip install pre-commit
pre-commit run -a
```

That will automatically run `black` for you.

**Comment by coloursofnoise** [comment]
Should be done, after a bit of confusion I realized that this branch doesn't have the latest changes to the `tox` and `pre-commit` config so I just ran `black` manually.

**Comment by caioariede** [comment]
@stephenfin any plans on releasing this to PyPI? 👀

**Pull request merged**